trace/context.go

```diff
@@ -16,9 +16,15 @@ package trace // import "go.opentelemetry.io/otel/trace"
 
 import "context"
 
-type traceContextKeyType int
+type contextKey string
 
-const currentSpanKey traceContextKeyType = iota
+// String returns the context key with a prefix to make the key globally unique
+// this prevents collisions with other packages
+func (c contextKey) String() string {
+	return "otel/trace" + string(c)
```

Comment:
It feels like it could be easy to miss what's happening when seeing a string `otel/tracecurrentSpan`. How about:

```suggestion
	return "otel/trace: " + string(c)
```

Then, we end up with `otel/trace: currentSpan`.